### PR TITLE
feat(NODE-4950)!: remove bson-ext import

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

Removes bson-ext import

#### What is the motivation for this change?

BSON-EXT no longer works with JS BSON v5 types

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/unit/bson.test.js [hunk 1]
```diff
@@ -47,37 +46,7 @@ describe('When importing BSON', function () {
...
-
-    it('should be imported by default', function () {
-      expect(BSON.deserialize.toString()).to.not.include('[native code]');
-      expect(BSON.serialize.toString()).to.not.include('[native code]');
-      expect(BSON.calculateObjectSize.toString()).to.not.include('[native code]');
-    });
-
-    testTypes();
-  });
+  testTypes();
```
Comment (durran) [inline test/unit/bson.test.js:49]: Can we just remove this function call and the function definition completely and just loop in the describe blocks to create the tests?

File: test/unit/commands.test.ts [hunk 2]
```diff
@@ -42,232 +41,105 @@ const nKeyWithInvalidUtf8 =
...
-      }
-    });
-
-    it('throws when invalid utf8 in writeErrors object regardless of options', () => {
-      // this is a sanity check to make sure nothing unexpected is happening in the deserialize method itself
-      const options = { validation: { utf8: { writeErrors: false } as const } };
-      const deserializerCall = () => BSON.deserialize(invalidUtf8ErrorMsgDeserializeInput, options);
-      expect(deserializerCall).to.throw(Error, 'Invalid UTF8 string found');
-    });
+  it('bson correctly deserializes data with replacement characters for invalid utf8 in writeErrors object', () => {
```
Comment (durran) [inline test/unit/commands.test.ts:44]:
```suggestion
  it('correctly deserializes data with replacement characters for invalid utf8 in writeErrors object', () => {
```

File: test/unit/commands.test.ts [hunk 3]
```diff
@@ -42,232 +41,105 @@ const nKeyWithInvalidUtf8 =
...
-        );
-      });
+    const options = {};
+    it('does not validate the writeErrors key', () => {
+      expect(() => binMsgInvalidUtf8ErrorMsg.parse(options)).to.not.throw();
     });
 
-    context('when validation is disabled', () => {
-      const binMsgInvalidUtf8ErrorMsg = new BinMsg(
+    it('should validate keys other than the writeErrors key', () => {
```
Comment (durran) [inline test/unit/commands.test.ts:64]:
```suggestion
    it('validates keys other than the writeErrors key', () => {
```

File: test/unit/commands.test.ts [hunk 4]
```diff
@@ -42,232 +41,105 @@ const nKeyWithInvalidUtf8 =
...
-        );
-      });
+    const options = { enableUtf8Validation: false };
+    it('should not validate the writeErrors key', () => {
+      expect(() => binMsgInvalidUtf8ErrorMsg.parse(options)).to.not.throw();
     });
 
-    it('should still validate for writeErrors if no validation specified', () => {
-      const binMsgInvalidUtf8ErrorMsg = new BinMsg(
+    it('should not validate keys other than the writeErrors key', () => {
```
Comment (durran) [inline test/unit/commands.test.ts:89]:
```suggestion
    it('does not validate keys other than the writeErrors key', () => {
```

File: test/unit/commands.test.ts [hunk 5]
```diff
@@ -42,232 +41,105 @@ const nKeyWithInvalidUtf8 =
...
-    });
-
-    it('bson correctly deserializes data with replacement characters for invalid utf8 in writeErrors object', () => {
-      // this is a sanity check to make sure nothing unexpected is happening in the deserialize method itself
-
-      const options = { validation: { utf8: { writeErrors: false } as const } };
-      const deserializerCall = () => BSON.deserialize(invalidUtf8ErrorMsgDeserializeInput, options);
-      expect(deserializerCall()).to.deep.equals(invalidUtf8InWriteErrorsJSON);
-    });
+  it('should by default disable validation for writeErrors if no validation specified', () => {
```
Comment (durran) [inline test/unit/commands.test.ts:102]:
```suggestion
  it('disables validation by default for writeErrors if no validation specified', () => {
```

File: test/unit/commands.test.ts [hunk 6]
```diff
@@ -42,232 +41,105 @@ const nKeyWithInvalidUtf8 =
...
+      promoteValues: true
+    };
+
+    expect(() => binMsgInvalidUtf8ErrorMsg.parse(options)).to.not.throw();
+  });
 
-    context('when enableUtf8Validation option is not specified', () => {
+  context('utf8 validation enabled', () => {
+    const options = { enableUtf8Validation: true };
+    it('should not validate the writeErrors key', () => {
```
Comment (durran) [inline test/unit/commands.test.ts:120]:
```suggestion
    it('does not validate the writeErrors key', () => {
```

File: test/unit/commands.test.ts [hunk 7]
```diff
@@ -42,232 +41,105 @@ const nKeyWithInvalidUtf8 =
...
-        expect(() => binMsgAnotherKeyWithInvalidUtf8.parse(options)).to.not.throw(
-          BSONError,
-          'Invalid UTF-8 string in BSON document'
-        );
-      });
     });
 
-    it('should by default disable validation for writeErrors if no validation specified', () => {
-      const binMsgInvalidUtf8ErrorMsg = new BinMsg(
+    it('should validate keys other than the writeErrors key', () => {
```
Comment (durran) [inline test/unit/commands.test.ts:133]:
```suggestion
    it('validates keys other than the writeErrors key', () => {
```

**Review by durran — CHANGES_REQUESTED** [review]

File: test/unit/bson.test.js (hunk 1 above)
Comment (nbbeeken) [reply test/unit/bson.test.js]: good clean up, done!

**Review by durran — DISMISSED** [review]

File: README.md [hunk 8]
```diff
@@ -73,12 +73,13 @@ npm install -D @types/node
...
 ## Troubleshooting
 
-The MongoDB driver depends on several other packages. These are:
+The MongoDB driver can optionally be enhanced by the following feature packages:
 
-- [bson](https://github.com/mongodb/js-bson)
-- [bson-ext](https://github.com/mongodb-js/bson-ext)
-- [kerberos](https://github.com/mongodb-js/kerberos)
-- [mongodb-client-encryption](https://github.com/mongodb/libmongocrypt#readme)
+- [Zstd network compression](https://github.com/mongodb-js/zstd)
```
Comment (dariakp) [inline README.md:78]: I think listing the package names is still worthwhile, if we want to add a short description like we have currently, we can do that, too, but I think it's more convenient for users to reference them by the name they can `npm install` (or look up in the npm repository). I am also not sure that we should be listing packages we don't own, and if we do list them, then we should clearly label them as 3rd party. Given that they are in the troubleshooting section, I'm leaning toward not including them at all.

**Review by dariakp — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (nbbeeken) [reply README.md]: Yea this section was a bit off, bson was listed and it's not a c++ dep nor is it something that needs troubleshooting (I hope 😅) 

Updated this section, and I put the link to the troubleshooting under our dependencies only

**Review by durran — APPROVED** [review]

**Review by dariakp — APPROVED** [review]

**Pull request merged**